dss/stepfunctions/__init__.py

```diff
@@ -16,18 +54,84 @@ def step_functions_invoke(state_machine_name_template: str, execution_name: str,
     """
 
     sfn = boto3.client('stepfunctions')
-    sts_client = boto3.client("sts")
 
     execution_input = json.dumps(input)
+    state_machine_arn = step_functions_arn(state_machine_name_template)
 
-    region = boto3.Session().region_name
-    accountid = sts_client.get_caller_identity()['Account']
-    stage = os.environ["DSS_DEPLOYMENT_STAGE"]
-    sfn_name = state_machine_name_template.format(stage=stage)
-    state_machine_arn = f"arn:aws:states:{region}:{accountid}:stateMachine:{sfn_name}"
     response = sfn.start_execution(
         stateMachineArn=state_machine_arn,
         name=execution_name,
         input=execution_input
     )
     return response
+
+
+def step_functions_describe_execution(state_machine_name_template: str, execution_name: str) -> typing.Any:
+    """
+    Return description of a step function exectuion, possible in-progress, completed, errored, etc.
+    :param state_machine_name_template:
+    :param execution_name:
+    :return:
+    """
+
+    sfn = boto3.client('stepfunctions')
+
+    execution_arn = step_functions_execution_arn(state_machine_name_template, execution_name)
+
+    resp = sfn.describe_execution(
+        executionArn=execution_arn
+    )
+
+    return resp
+
+
+def step_functions_list_executions(state_machine_name_template: str,
+                                   start_date: typing.Type(datetime.datetime)=None,
+                                   k_results_per_page: int=100,
+                                   k_max_pages: int=50):
+    """
+    List executions of a step function earlier than start_date, performing paged api calls in background.
+    Maximum 100 results per page.
+    :param state_machine_name_template:
+    :param k_results_per_page:
+    :param start_date:
+    :param k_max_pages:
+    :return:
+    """
+
+    if start_date is None:
+        start_date = datetime.datetime(1970, 1, 1).astimezone()
+
+    sfn = boto3.client('stepfunctions')
+
+    state_machine_arn = step_functions_arn(state_machine_name_template)
+
+    if k_max_pages < 1:
+        raise Exception('Need at least 1 AWS api call')
+
+    executions = list()
+
+    kwargs = {
+        'stateMachineArn': state_machine_arn,
+        'maxResults': k_results_per_page
+    }
+
+    for k_api_calls in range(1, k_max_pages + 1):
+        resp = sfn.list_executions(**kwargs)
+
+        execs = resp['executions']
+
+        if execs:
+            if execs[-1]['startDate'] < start_date:
+                executions.extend([e for e in execs
+                                   if e['startDate'] > start_date])
+                break
+            else:
+                executions.extend(execs)
+
+        if not resp.get('nextToken', False):
+            break
+
```

Comment:
I would add `else:` here to be consistent and explicit.

And then I would remove `not` and swap the branches.